### ISSUE TITLE
calling Model in a loop would leak memory

### ISSUE THREAD

**Issue by alerem18** [tensorflow/tensorflow#61942]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

tf2.13.0

### Custom code

Yes

### OS platform and distribution

Windows Server 2019

### Current behavior?

there is a transformer model when i try to decode messages, translate input sentence to target sentence i get memory blow up, memory is good when i use transformer.fit(), but in a loop like below it blows up memory, tf.keras.backend.clear_session() does’t help, also accuracy decrease when i use that, gc.collect() doesn’t work also
here is my code

`from tqdm import tqdm
def overall_accuracy(pairs):
    corrects = 0
    inputs = pairs[2739:]
    iter = tqdm(inputs)
    for i, pair in enumerate(iter):
        input_text = pair[0]
        target = pair[1]
        predicted = decode_sequence(input_text)
        #guess = '✓' if predicted == target else '✗'
        #print('Sample Number : ', i, 'Predicted : ', predicted, 'Real : ', target, guess)
        if predicted == target:
            corrects += 1
        iter.set_postfix(corrects=corrects, accuracy=corrects / (i + 1))
    
    return corrects / len(inputs)

print("Overall Acurracy : ", overall_accuracy(test_pairs))

from tqdm import tqdm
def overall_accuracy(pairs):
    corrects = 0
    inputs = pairs[2739:]
    iter = tqdm(inputs)
    for i, pair in enumerate(iter):
        input_text = pair[0]
        target = pair[1]
        predicted = decode_sequence(input_text)
        #guess = '✓' if predicted == target else '✗'
        #print('Sample Number : ', i, 'Predicted : ', predicted, 'Real : ', target, guess)
        if predicted == target:
            corrects += 1
        iter.set_postfix(corrects=corrects, accuracy=corrects / (i + 1))
    
    return corrects / len(inputs)

print("Overall Acurracy : ", overall_accuracy(test_pairs))`

### Standalone code to reproduce the issue

```shell
calling model in a loop
```

**Comment by sushreebarsa** [comment]
@alerem18 Could you please provide the complete standalone code to replicate the [issue](https://colab.research.google.com/gist/sushreebarsa/1312d871663ec6136f188bc447d4b939/61942.ipynb) reported?
Thank you!

**Comment by alerem18** [comment]
https://keras.io/examples/nlp/neural_machine_translation_with_transformer/

in decoder_sequence

**Comment by alerem18** [comment]
will there be any solution or i should wait for you to reply once a month?
this is why people prefer pytorch over tensorflow

**Comment by sushreebarsa** [comment]
@alerem18 I tried to replicate the issue on colab. https://keras.io/examples/nlp/neural_machine_translation_with_transformer/ is working fine in [colab](https://colab.research.google.com/gist/sushreebarsa/e8ca05c13b4748b40911c8c00a86a7bc/neural_machine_translation_with_transformer.ipynb#scrollTo=7Wz8hlIULZul) . Could you please try to use the GPU for this issue and let me know if the memory leak would be there? Thank you!

**Comment by alerem18** [comment]
> @alerem18 I tried to replicate the issue on colab. https://keras.io/examples/nlp/neural_machine_translation_with_transformer/ is working fine in [colab](https://colab.research.google.com/gist/sushreebarsa/e8ca05c13b4748b40911c8c00a86a7bc/neural_machine_translation_with_transformer.ipynb#scrollTo=7Wz8hlIULZul) . Please let me know if i have missed something to replicate the reported error. Have you tried to use the GPU for this issue and let me know if the memory leak would be there? Thank you!

i've not tested it on gpu, but i have memory issues when translating a lot of texts, in a loop

**Comment by alerem18** [comment]
@sachinprasadhs same issue on this
https://www.tensorflow.org/text/tutorials/transformer

**Comment by alerem18** [comment]
is there any damn solution for this or not?

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
